src/containers/forms/OrganizationForm/index.tsx

```diff
@@ -0,0 +1,195 @@
+import { ErrorMessage, Field, Form, Formik } from 'formik';
+import moment from 'moment';
+import React, { useState } from 'react';
+import { Redirect } from 'react-router';
+import { Button, Label } from 'reactstrap';
+import { FormGroup } from 'reactstrap';
+import * as Yup from 'yup';
+import {
+  ACTIVE,
+  NAME,
+  NO,
+  OPENSRP_ORGANIZATION_ENDPOINT,
+  ORGANIZATION_LABEL,
+  ORGANIZATIONS_LIST_URL,
+  REQUIRED,
+  SAVE,
+  SAVING,
+  YES,
+} from '../../../constants';
+import { generateNameSpacedUUID } from '../../../helpers/utils';
+import { OpenSRPService } from '../../../services/opensrp';
+
+const OrgFormNameSpace = '9a4c8cb0-df70-11e9-b38b-57f114a50538';
+
+/** default type value for organizations */
+const defaultOrganizationType = {
+  coding: [
+    {
+      code: 'team',
+      display: 'Team',
+      system: 'http://terminology.hl7.org/CodeSystem/organization-type',
+    },
+  ],
+};
+
+/** yup validation schema for teams Form input */
+export const OrgSchema = Yup.object().shape({
+  active: Yup.boolean(),
+  identifier: Yup.string(),
+  name: Yup.string().required(REQUIRED),
+});
+
+/** interface for data fields for team's form */
+export interface OrganizationFormFields {
+  active: boolean;
+  identifier: string;
+  name: string;
+  type: typeof defaultOrganizationType;
+}
+
+/** interface for Organization form props */
+export interface OrganizationFormProps {
+  disabledFields: string[];
+  initialValues: OrganizationFormFields;
+  redirectAfterAction: string;
+}
+
+export const defaultInitialValues: OrganizationFormFields = {
+  active: false,
+  identifier: '',
+  name: '',
+  type: defaultOrganizationType,
+};
+
+// TODO - indicator that a team has been added to store??
+/** Organization form component */
+const OrganizationForm = (props: OrganizationFormProps) => {
+  /** track when redirection from this form page should occur */
+  const [ifDoneHere, setIfDoneHere] = useState<boolean>(false);
+  const { initialValues, redirectAfterAction, disabledFields } = props;
+  const [globalError, setGlobalError] = useState<string>('');
+
+  /** edit mode set to true if we are updating a team data. */
+  const editMode: boolean = initialValues.identifier !== '';
+
+  return (
+    <div className="form-container">
+      {ifDoneHere && <Redirect to={redirectAfterAction} />}
+      <Formik
+        initialValues={initialValues}
+        validationSchema={OrgSchema}
+        // tslint:disable-next-line: jsx-no-lambda
+        onSubmit={(values, { setSubmitting, setFieldValue }) => {
+          const organizationService = new OpenSRPService(OPENSRP_ORGANIZATION_ENDPOINT);
+          setFieldValue(
+            'identifier',
+            generateNameSpacedUUID(`${moment().toString()}`, OrgFormNameSpace)
+          );
```

Comment:
@p-netm does this happen synchronously? When submitting I'm getting an error from OpenSRP because the `identifier` was not provided (it's an empty string in the request).

Is it better to spread the values and set the identifier manually here instead?

Reply:
This is resolved by 7a1635a